config/provision/site-cookbooks/web_server/recipes/default.rb

```diff
@@ -0,0 +1,269 @@
+iptables_ng_rule '04-http' do
+  rule '-p tcp --dport 80 -j ACCEPT'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '05-https' do
+  rule '-p tcp --dport 443 -j ACCEPT'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '06-ssh' do
+  rule '-p tcp -m state --state NEW --dport 22 -j ACCEPT'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '07-ping-v4' do
+  rule '-p icmp -m icmp --icmp-type 8 -j ACCEPT'
+  ip_version 4
+  action :create_if_missing
+end
+
+iptables_ng_rule '08-log' do
+  rule '-m limit --limit 5/min -j LOG --log-prefix "iptables denied: " --log-level 7'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '09-input-block' do
+  rule '-j DROP'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '10-input-ip6' do
+  rule '-j DROP'
+  action :create_if_missing
+  ip_version 6
+end
+
+iptables_ng_rule '11-forward-ip6' do
+  chain 'FORWARD'
+  rule '-j DROP'
+  action :create_if_missing
+  ip_version 6
+end
+
+iptables_ng_rule '12-output' do
+  chain 'OUTPUT'
+  rule '-j ACCEPT'
+  action :create_if_missing
+  ip_version 4
+end
+
+iptables_ng_rule '13-output-ip6' do
+  chain 'OUTPUT'
+  rule '-j DROP'
+  action :create_if_missing
+  ip_version 6
+end
+
+include_recipe "chruby_install"
+include_recipe "ruby-install::install"
+
+directory app_root do
+  owner app_user
+  group app_user
+  recursive true
+  mode '0755'
+end
+
+directory "#{app_root}/shared" do
+  owner app_user
+  group app_user
+  mode '0755'
+end
+
+%w( config log tmp sockets pids ).each do |dir|
+  directory "#{app_root}/shared/#{dir}" do
+    owner app_user
+    group app_user
+    mode '0755'
+  end
+end
+
+database_name = node['web_server']['database']['name']
+database_user = node['web_server']['database']['user']
+database_password = Chef::EncryptedDataBagItem.load("web_server", "database")['password']
+template "#{app_root}/shared/config/database.yml" do
+  mode '0640'
+  owner app_user
+  group app_user
+  source "database.yml.erb"
+  variables({
+    database: database_name,
+    username: database_user,
+    password: database_password
+  })
+end
+
+secret_key = Chef::EncryptedDataBagItem.load("web_server", "secrets")['secret_key']
+template "#{app_root}/shared/config/secrets.yml" do
+  mode '0640'
+  owner app_user
+  group app_user
+  source "secrets.yml.erb"
+  variables({
+    secret_key: secret_key
+  })
+end
+
+paperclip_secret = Chef::EncryptedDataBagItem.load("web_server", "application_config")['paperclip_secret']
+template "#{app_root}/shared/config/application_config.yml" do
+  mode '0640'
+  owner app_user
+  group app_user
+  source "application_config.yml.erb"
+  variables({
+    paperclip_secret: paperclip_secret
+  })
+end
+
+puma_config app_name do
+  owner app_user
+  directory app_root
+  upstart true
+  monit false
+  logrotate true # the default
+  thread_min 0
+  thread_max 16
+  workers 2
+end
+
+# Puma init
+template "/etc/init/puma.conf" do
```

Comment:
This template is for upstart which is not used in the latest Ubuntu.  Puma provides us with a basic systemd template here: https://github.com/puma/puma/blob/master/docs/systemd.md